src/components/filters/input.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from "react"
+
+import Input from "../input"
+
+import { scrollIntoViewIfMobile } from "../../lib/scrollHelper"
+import getClosestElement from "../../lib/getClosestElement"
+
+interface Props {
+  name: string
+  initialValue?: string
+  mode: "text" | "numeric" | "decimal" | "tel" | "email"
+  placeholder: string
+  step?: number
+  min?: number
+  max?: number
+  position?: "left" | "right"
+  apply: (filtersToApply) => void
+}
+
+interface State {
+  refocus: boolean
+  value: string
+}
+
+const timeout = 1000
+
+export default class InputFilter extends Component<Props, State> {
+  state = {
+    refocus: false,
+    value: this.props.initialValue,
+  }
+
+  timeoutFn = null
+
+  onChange = (event) => {
+    const value = event.target.value
+    this.setState({ value })
+    if (event.target.cleared) {
+      this.props.apply({ [this.props.name]: value })
+    }
+  }
+
+  onKeyDown = (event) => {
+    clearTimeout(this.timeoutFn)
+    event.persist()
+    const value = event.target.value
+
+    if (event.keyCode === 13) {
+      event.preventDefault()
+      this.props.apply({ [this.props.name]: value })
+    } else {
+      this.timeoutFn = setTimeout(
+        () => this.props.apply({ [this.props.name]: value }),
+        timeout
+      )
+    }
+  }
+
+  onFocus = (event) => {
+    event.persist()
+    this.setState({ refocus: true }, () => {
+      const closestSection = getClosestElement(
+        event.target,
+        "[data-closestpoint]"
+      )
+      scrollIntoViewIfMobile(closestSection)
```

Comment:
would it be possible to use a ref instead?

Reply:
Given this component is quite complex I would not refactor it too heavily in the scope of this PR. I'd be happy to tackle this separately.

Reply:
yeah, didn't realize you're just porting!